### ISSUE TITLE
Switching from speaker to loudspeaker is very glitchy.

### ISSUE THREAD

**Issue by mudassirzulfiqar** [VoIPGRID/VialerSIPLib#237]
### Version

Latest

### File / Feature

Change output to LoudSpeaker

### Expected behavior

It should smoothly switch to loudspeaker and vice versa

### Actual behavior

The other person hears a very glitchy voice , and its very difficult to understand any thing, and it remains this way for 8 seconds right after switch

### Stacktrace / Error message

No, it has been observed

### Other info

I have tested many client apps for iOS, many of them also have the same problem , I search a lot on about it. I thinks its more like an acho cancelation problem, I tried all Echo Flags in this library but nothing changed. On Android our sip client works fine.

**Comment by ismaiI1** [comment]
That's right.. When I changed the voice device, sound is being robotic 8-10 sec..

**Comment by mudassirzulfiqar** [comment]
@invisible66 exactly same thing.

**Comment by mudassirzulfiqar** [comment]
@invisible66 any solution ?

**Comment by ismaiI1** [comment]
Unfortunately no..
